kitsune/users/api.py

```diff
@@ -365,3 +369,41 @@ def delete_setting(self, request, user__username=None):
             return Response(status=status.HTTP_204_NO_CONTENT)
         except Setting.DoesNotExist:
             raise GenericAPIException(404, {'detail': 'No matching user setting found.'})
+
+    @action(methods=['GET'])
+    def request_password_reset(self, request, user__username=None):
+        profile = self.get_object()
+
+        current_site = get_current_site(request)
+        site_name = current_site.name
+        domain = current_site.domain
+
+        c = {
+            'email': profile.user.email,
+            'domain': domain,
+            'site_name': site_name,
+            'uid': int_to_base36(profile.user.id),
+            'user': profile.user,
+            'token': default_token_generator.make_token(profile.user),
+            'protocol': 'https' if request.is_secure() else 'http',
+        }
+
+        subject = email_utils.render_email('users/email/pw_reset_subject.ltxt', c)
+        # Email subject *must not* contain newlines
+        subject = ''.join(subject.splitlines())
+
+        @email_utils.safe_translation
+        def _make_mail(locale):
+            mail = email_utils.make_mail(
+                subject=subject,
+                text_template='users/email/pw_reset.ltxt',
+                html_template='users/email/pw_reset.html',
+                context_vars=c,
+                from_email=None,
+                to_email=profile.user.email)
+
+            return mail
+
+        email_utils.send_messages([_make_mail(profile.locale)])
+
+        return Response({'success': True})
```

Comment:
The idiom I've been using for this is `return Response('', status=204)`. 204 means "success, but no body". I think it would be could to be consistent.